doc/topics/core-structure.md

```diff
@@ -0,0 +1,202 @@
+# Node Core's Internal Structure
+
+This document covers the basics of how Node.js is structured, in an attempt to
+make it easier to contribute to.
+
+Note: The Node.js codebase is in a constant amount of flux, things can and will
+change over time, and so also parts of this document will go out of date over
+time.
+
+
+## Top-Level File & Folder Structure
+
+At the root of Node.js you have a set of files and folders that can be roughly
+identified as the following:
+
+- CAPITALS documentation files
+- Documentation (`doc/`)
+- Top-level Build Tooling
+- JavaScript "standard" Library (`lib/`)
+- C++ "native" Source (`src/`)
+- Tests & Benchmarks (`test/` & `benchmark/`)
+- Internal / Build Tooling (`tools/`)
+- Dependencies (`deps/`)
+- .dotfiles
+
+Below is a summary of the parts, ordered as per the list above.
+
+
+### CAPITALS files
+
+These are pretty important, and describe things such as our [GOVERNANCE][], [CONTRIBUTING][], and [CODE_OF_CONDUCT][] policies.
```

Comment:
Consider removing a bunch of unneeded words here: `These files describe our policies such as [GOVERNANCE][]...`